ClockDisplay.java

```diff
@@ -81,7 +88,85 @@ public String getTime()
      */
     private void updateDisplay()
     {
-        displayString = hours.getDisplayValue() + ":" + 
+        switch(hours.getValue())
+        {
+            case 1:
+                displayHours = 1;
+                break;
+            case 2:
+                displayHours = 2;
+                break;
+            case 3:
+                displayHours = 3;
+                break;
+            case 4:
+                displayHours = 4;
+                break;
+            case 5:
+                displayHours = 5;
+                break;
+            case 6:
+                displayHours = 6;
+                break;
+            case 7:
+                displayHours = 7;
+                break;
+            case 8:
+                displayHours = 8;
+                break;
+            case 9:
+                displayHours = 9;
+                break;
+            case 10:
+                displayHours = 10;
+                break;
+            case 11:
+                displayHours = 11;
+                break;
+            case 12:
+                displayHours = 12;
+                break;
+            case 13:
+                displayHours = 1;
+                break;
+            case 14:
+                displayHours = 2;
+                break;
+            case 15:
+                displayHours = 3;
+                break;
+            case 16:
+                displayHours = 4;
+                break;
+            case 17:
+                displayHours = 5;
+                break;
+            case 18:
+                displayHours = 6;
+                break;
+            case 19:
+                displayHours = 7;
+                break;
+            case 20:
+                displayHours = 8;
+                break;
+            case 21:
+                displayHours = 9;
+                break;
+            case 22:
+                displayHours = 10;
+                break;
+            case 23:
+                displayHours = 11;
+                break;
+            case 24:
+                displayHours = 12;
+                break;
+            default:
+                displayHours = 0;
+        }
```

Comment:
This does what is asked. You can also do

int tempHours = hours.getValue();
if (tempHours  >= 12) tempHours-= 12;
if (tempHours ==0) tempHours = 12;